File: 品味Java的21種設計模式/dp_cpp/command/example1/Invoker.cpp
#include "Invoker.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace command
			{
				namespace example1
				{

					void Invoker::setCommand(Command *command)
					{
						this->command = command;
					}

					void Invoker::runCommand()
					{
						//调用命令对象的执行方法
						command->execute();
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/command/example1/Invoker.java
package cn.javass.dp.command.example1;
/**
 * 调用者
 */
public class Invoker {
    /**
     * 持有命令对象
     */
    private Command command = null;
    /**
     * 设置调用者持有的命令对象
     * @param command 命令对象
     */
    public void setCommand(Command command) {
        this.command = command;
    }
    /**
     * 示意方法，要求命令执行请求
     */
    public void runCommand() {
        //调用命令对象的执行方法
        command.execute();
    }
}
